pkg/apis/core/validation/validation.go

```diff
@@ -3736,6 +3752,58 @@ func validatePodIPs(pod *core.Pod) field.ErrorList {
 	return allErrs
 }
 
+// validateHostIPs validates IPs in pod status
+func validateHostIPs(pod *core.Pod) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(pod.Status.HostIPs) == 0 {
+		return allErrs
+	}
+
+	hostIPsField := field.NewPath("status", "hostIPs")
+
+	// hostIP must be equal to hostIPs[0].IP
+	if pod.Status.HostIP != pod.Status.HostIPs[0].IP {
```

Comment:
I know I flagged this before, but I'm looking for where this is done for podIPs and not finding it.  I need to revisit this.

Reply:
In the original patch, I fully referenced `podIP` and `podIPs`, and I changed a lot after multiple rounds of review, but I also remember that `podIPs` didn't have that. But I think there should be no problem to add it in as well.

I don't have a hard opinion.

Reply:
I went digging through the history and we are not awesome at being consistent.

For service `clusterIPs` we are very careful to try to handle old clients and new clients.  If you set the new (plural) field you must ALSO set the old (singular) field or we will fail validation.  To do that we retained both fields on internal.  If an old client only sets `clusterIP`, we force `clusterIPs` to match.  If a new client setsThere was a reason I can't recall right now why we don't want new clients to only set the plural, but I can't recall the reason now, and it's not in the code.  @khenidak may recall why we decided that it was important for users of the plural to ALSO set the singular.

For pod `hostIPs` we use conversion and squash them together into just the plural field on internal.  If an old client only sets `hostIP`, we set `hostIPs` for them.  If a client only sets `hostIPs`, we set `hostIP` for them.  If a client sets BOTH, we silently discard the `hostIPs`, when it SHOULD be an error.  So retaining both is more correct.